sdk/include/opentelemetry/sdk/metrics/export/metric_producer.h

```diff
@@ -31,7 +31,7 @@ namespace metrics
  */
 struct ScopeMetrics
 {
-  const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope_;
+  const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope_ = nullptr;
```

Comment:
I think this change is the reason for the CI failures. Now that this field has an initial value, `ScopeMetrics` is no longer considered an "aggregate type", which [breaks](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/6275544529/job/17043256873?pr=2293#step:6:726) the way it's used in the OTLP tests:

```
ostream_metric_test.cc:58:70: error: no matching function for call to ‘std::vector<opentelemetry::v1::sdk::metrics::ScopeMetrics>::vector(<brace-enclosed initializer list>)’
       {scope.get(), std::vector<metric_sdk::MetricData>{metric_data}}};D
```

Reply:
I removed this, and that fixed the tests.